File: my-folder/0021-merge-two-sorted-lists/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode newList = new ListNode();
        ListNode head = newList;
        while (list1 != null && list2 != null) {
            if (list1.val > list2.val) {
                head.next = list2;
                head = head.next;
                list2 = list2.next;
            } else {
                head.next = list1;
                head = head.next;
                list1 = list1.next;
            }
        }
        if (list1 == null && list2 == null) {
            return newList.next;
        } else  if (list1 == null) {
            head.next = list2;
            return newList.next;
        } else {
            head.next = list1;
            return newList.next;
        }
        
    }
}


/*

2   4
3   5

newList () -> 1124
head           . 
*/

File: my-folder/0021-merge-two-sorted-lists/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution(object):
    def mergeTwoLists(self, list1, list2):
        """
        :type list1: Optional[ListNode]
        :type list2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        
        new_list = ListNode()
        start = new_list
        while list1 != None and list2 != None:
            if list1.val < list2.val:
                new_list.next = list1
                new_list = new_list.next
                list1 = list1.next
            else:
                new_list.next = list2
                new_list = new_list.next
                list2 = list2.next
        if list1 != None:
            new_list.next = list1
        elif list2 != None:
            new_list.next = list2
        return start.next
